### PR TITLE
[client/rest]: add NEM OperationParser

### PR DESCRIPTION
**Pull request by Jaguar0625**
[client/rest]: add NEM OperationParser
    
     problem: need to convert NEM transactions to rosetta operations
    solution: add NEM OperationParser

    [client/rest]: promote RosettaOperationFactory to shared test utils
    
     problem: both NEM and Symbol tests need RosettaOperationFactory
    solution: promote into shared test utils

### CODE REVIEW

File: client/rest/src/plugins/rosetta/nem/OperationParser.js [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	async parseTransactionInternal(transaction) {
+		const operations = [];
+
+		const pushDebitCreditOperations = (sourcePublicKey, targetAddress, amount, currency) => {
+			operations.push(this.createDebitOperation({ sourcePublicKey, amount, currency }));
+			operations.push(this.createCreditOperation({ targetAddress, amount, currency }));
+		};
+
+		const pushRegistrationFeeOperation = async () => {
+			const amount = transaction.rentalFee;
```
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/nem/OperationParser.js:275]: Should ``rentalFee`` be ``creationFee`` for mosaics definition tx?
https://nemproject.github.io/#namespaces-and-mosaics

**Review by Wayonb — APPROVED** [review]
👍🏾

File: client/rest/src/plugins/rosetta/nem/OperationParser.js (hunk 1 above)
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/OperationParser.js]: yea, you're right. updated.

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**